File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.java
class Solution {
    public int maxDepth(String s) {
        int level=0,maxLevel=0;
        for(char ch:s.toCharArray()){
            if(ch=='('){
                level++;
            }else if(ch==')'){
                maxLevel=Math.max(maxLevel,level);
                level--;
            }
        }
        return maxLevel;
    }
}

File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        int cnt=0,maxCnt=0;
        for(char c:s){
            if(c=='('){
                cnt++;
            }else if(c==')'){
                maxCnt=max(cnt,maxCnt);
                cnt--;
            }
        }
        return maxCnt;
    }
};